File: c++/template.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

#if _EDITOR

#include "PROBLEM_NUMBER.h"

static const char * fsc_pChzName = "PROBLEM_NAME";
static const char * fsc_pChzURL = "https://www.codeeval.com/open_challenges/PROBLEM_NUMBER/";
static const int fsc_nNumber = PROBLEM_NUMBER;
static const bool fsc_fFinished = false;
static const EDifficulty fsc_eDifficulty = PROBLEM_DIFFICULTY;

CPPROBLEM_NUMBER::CPPROBLEM_NUMBER()
	: super(string(fsc_pChzName), string(fsc_pChzURL), fsc_nNumber, fsc_fFinished, fsc_eDifficulty)
{
}

CPPROBLEM_NUMBER::~CPPROBLEM_NUMBER()
{
}

int CPPROBLEM_NUMBER::nRun(int argc, const char * argv[])
#else
int main(int argc, char* argv[])
#endif
{
	fstream t_file(argv[1], ios::in);
	
	string t_strInput;

	while (getline(t_file, t_strInput))
	{
		
	}

	t_file.close();

	return 0;
}

File: c#/template.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
	
namespace codeeval
{
#if _EDITOR
	public class CPPROBLEM_NUMBER : CProblem
	{
#else
	class Problem
	{
#endif

#if _EDITOR
		private const string mc_strName = "PROBLEM_NAME";
		private const string mc_strURL = "https://www.codeeval.com/open_challenges/PROBLEM_NUMBER/";
		private const int mc_nNumber = PROBLEM_NUMBER;
		private const bool mc_fFinished = false;
		private const EDifficulty mc_eDifficulty = EDifficulty.PROBLEM_DIFFICULTY;

		public CPPROBLEM_NUMBER() : base(mc_strName, mc_strURL, mc_nNumber, mc_fFinished, mc_eDifficulty)
		{
		}

		public override void vRun()
		{
			StreamReader t_reader = File.OpenText("in/PROBLEM_NUMBER_in.txt");
#else
		static void Main(string[] args)
		{
			StreamReader t_reader = File.OpenText(args[0]);
#endif
				
			while (!t_reader.EndOfStream)
			{
				string t_line = t_reader.ReadLine();
				if (t_line == null)
				{
					continue;
				}
			}
		}
	}
}
